### PR TITLE
protocol 8.

### PR DESCRIPTION
**Pull request by cloudwebrtc**
I did some tests, we can connect the room in the following two ways

```dart
      final room = Room();
      final listener = room.createListener();

```

a)
``` dart
      await room.connect(
        _url,
        _token,
        roomOptions: RoomOptions(
           ...
        ),
        fastConnectOptions: _fastConnect
            ? FastConnectOptions(
                microphone: const TrackOption(enabled: true),
                camera: const TrackOption(enabled: true),
              )
            : null,
      );

```

b)
```dart
     final localAudioTrack = await LocalAudioTrack.create();
     final localCamTrack = await LocalVideoTrack.createCameraTrack();

      await room.connect(
        _url,
        _token,
        roomOptions: RoomOptions(
           ...
        ),
        fastConnectOptions: _fastConnect
            ? FastConnectOptions(
                microphone: TrackOption(track: localAudioTrack),
                camera:  TrackOption(track: localCamTrack),
              )
            : null,
      );

```

```

### CODE REVIEW

File: lib/src/core/room.dart [hunk 1]
```diff
@@ -141,6 +145,32 @@ class Room extends DisposableChangeNotifier with EventsEmittable<RoomEvent> {
         info: event.response.participant,
       );
 
+      if (connectOptions.protocolVersion.index >= ProtocolVersion.v8.index &&
```
Comment (davidliu) [inline lib/src/core/room.dart:148]: nit: probably should be moved elsewhere, since this is for listener setup.

**Review by davidliu — APPROVED** [review]

File: lib/src/livekit.dart [hunk 2]
```diff
@@ -9,6 +9,8 @@ class LiveKitClient {
   /// Convenience method for connecting to a LiveKit server.
   /// Returns a [Room] upon a successful connect or throws when it fails.
   /// Alternatively, it is possible to instantiate [Room] and call [Room.connect] directly.
+  @Deprecated(
+      'Use `Room.connect` instead, This method is deprecated above Protocol v8.')
   static Future<Room> connect(
     String url,
     String token, {
```
Comment (davidliu) [inline lib/src/livekit.dart:16]: Can we to default to protocol 7 when using the deprecated connect?

Comment (cloudwebrtc) [reply lib/src/livekit.dart:16]: ah, let me check this.

**Review by boks1971 — APPROVED** [review]

File: lib/src/livekit.dart (hunk 2 above)
Comment (cloudwebrtc) [reply lib/src/livekit.dart:18]: ok, solved

**Pull request merged**